### PR TITLE
add quantized GELU activation

### PR DESCRIPTION
**Pull request by vkuzo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#34396 add quantized GELU activation**

Summary:

Adds quantized implementation of GELU. Since the floating point implementation
doesn't currently have the inplace and out flavors, we also skip them here for
now.

Would love reviewers' feedback on if we need to do anything smarter with
scale+zp here, especially around -2 < x < 0.  Will address that either
as followup on this PR or as a separate PR, depending of what people
say and whether this is important.

Test Plan:

```
python test/test_quantized.py
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20371751](https://our.internmc.facebook.com/intern/diff/D20371751)

### CODE REVIEW

File: test/test_quantized.py [hunk 1]
```diff
@@ -169,6 +170,31 @@ def test_qrelu_leaky(self, X, alpha):
...
+        X, (scale, zero_point, torch_type) = X
+
+        X = torch.from_numpy(X)
+        qX = torch.quantize_per_tensor(X, scale=scale, zero_point=zero_point,
+                                       dtype=torch_type)
+
+        # calculate GT
+        dqX = qX.dequantize()
+        # GELU
+        dqY_hat = dqX * 0.5 * (1.0 + (dqX / math.sqrt(2)).erf())
```
Comment (z-a-f) [inline test/test_quantized.py]: Isn't it better to use the floating point version of the gelu here (`nn.functional.gelu`).

**Review by z-a-f — APPROVED** [review]
LGTM + nit

File: test/test_quantized.py [hunk 2]
```diff
@@ -169,6 +170,31 @@ def test_qrelu_leaky(self, X, alpha):
...
+                       qparams=hu.qparams()),
+           alpha=st.floats(0.0, 1.0, allow_nan=False, allow_infinity=False))
+    def test_qgelu(self, X, alpha):
+        X, (scale, zero_point, torch_type) = X
+
+        X = torch.from_numpy(X)
+        qX = torch.quantize_per_tensor(X, scale=scale, zero_point=zero_point,
+                                       dtype=torch_type)
+
+        # calculate GT
```
Comment (supriyar) [inline test/test_quantized.py]: nit: Not sure what you mean by GT here

**Comment by raghuramank100** [comment]
Can we also add benchmark results showing the performance improvement of the quantized impl over float?

**Comment by vkuzo** [comment]
> Can we also add benchmark results showing the performance improvement of the quantized impl over float?

will do, need to chase down the real test failure first as seems like the math is wrong in some CircleCI environments

**Comment by vkuzo** [comment]
There are some flaky test failures here which need to be fixed before this can land.  My best guess is that `std::erf` is misbehaving on some platforms.  Closing for now since there is no immediate ask for this, but we will revisit when needed.

**Pull request closed without merging**